src/menu/ExportIEDParams.ts

```diff
@@ -0,0 +1,303 @@
+      return typeElement.querySelector(`:scope > DO[name="${name}"]`);
+    } else if (typeElement.tagName === 'DOType') {
+      return typeElement.querySelector(
+        `:scope > SDO[name="${name}"], :scope > DA[name="${name}"]`
+      );
+    } else {
+      return typeElement.querySelector(`:scope > BDA[name="${name}"]`);
+    }
+  }
+
+  /**
+   * Retrieve the value that will be added to the CSV file. If an attribute name is passed the value of the
+   * attribute is returned. Otherwise, the textContent of the element is returned.
+   * @param element       - The element to retrieve the value from.
+   * @param attributeName - Optional the name of the attribute.
+   */
+  private getValue(
+    element: Element,
+    attributeName: string | undefined
+  ): string {
+    if (attributeName) {
+      return element.getAttribute(attributeName) ?? '';
+    }
+    return element.textContent ?? '';
+  }
+
+  /**
+   * Use the DO/SDO/DA/BDA data element to search for the type element. In case of the DO/SDO a DOType is search
+   * for and otherwise a DAType is searched for if the data element is a struct type.
+   *
+   * @param lastElement - The data element to retrieve its type definition.
+   */
+  private getTypeElement(lastElement: Element | null): Element | null {
+    if (lastElement) {
+      if (['DO', 'SDO'].includes(lastElement.tagName)) {
+        const type = lastElement.getAttribute('type') ?? '';
+        return this.doc.querySelector(`DOType[id="${type}"]`);
+      } else {
+        const bType = lastElement.getAttribute('bType') ?? '';
+        if (bType === 'Struct') {
+          const type = lastElement.getAttribute('type') ?? '';
+          return this.doc.querySelector(`DAType[id="${type}"]`);
+        }
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Search for the DO/SDO/DA/BDA element in the Template section of the document using the path array passed.
+   * The LN element is the starting point for the search in the Template section.
+   *
+   * @param lnElement         - The LN Element used as starting point in the Template section.
+   * @param dataAttributePath - The list of elements to search for, the names of the elements.
+   */
+  private getDataAttributeTemplateValue(
+    lnElement: Element,
+    dataAttributePath: string[]
+  ): string | null {
+    // This is only useful if the element to start from is the LN(0) Element.
+    if (
+      ['LN', 'LN0'].includes(lnElement.tagName) &&
+      dataAttributePath.length >= 2
+    ) {
+      // Search LNodeType Element that is linked to the LN(0) Element.
+      const type = lnElement.getAttribute('lnType');
+      let typeElement = this.doc.querySelector(`LNodeType[id="${type}"]`);
+      let lastElement: Element | null = null;
+
+      // Now start search through the Template section jumping between the type elements.
+      dataAttributePath.forEach(name => {
+        if (typeElement) {
+          lastElement = this.getDataElement(typeElement, name);
```

Comment:
This is the only part where the names are a bit confusing. Maybe something a bit more descriptive would be 
lastElement -> leafElement
getDataElement -> getDataTypeChildElement
typeElement -> dataTypeTemplate

Reply:
Always hard to find good names for this, but these look better, changed them.